gearcmd/worker.go

```diff
@@ -114,6 +114,9 @@ func (conf TaskConfig) doProcess(job baseworker.Job) error {
 		// Will be nil if the channel was closed without any errors
 		return err
 	case <-time.After(conf.CmdTimeout):
+		if err := cmd.Process.Kill(); err != nil {
```

Comment:
The processes spawned by gearcmd aren't long-running, so it doesn't pass the signals along. One process is one job, so gearcmd handles the signals itself by waiting for the current process to finish, and then not accepting any more jobs. This is handled by baseworker-go, a library used by gearcmd.